ipld/retriever.go

```diff
@@ -131,29 +126,28 @@ func (r *Retriever) newSession(ctx context.Context, dah *da.DataAvailabilityHead
 		r.bServ,
 		format.MaxSizeBatchOption(batchSize(size)),
 	)
-	ses := &retrievalSession{
-		bget:  blockservice.NewSession(ctx, r.bServ),
-		adder: adder,
-		treeFn: func() rsmt2d.Tree {
-			tree := wrapper.NewErasuredNamespacedMerkleTree(uint64(size)/2, nmt.NodeVisitor(adder.Visit))
-			return &tree
-		},
-		codec:     DefaultRSMT2DCodec(),
-		dah:       dah,
-		quadrants: newQuadrants(dah),
-		sharesLks: make([]sync.Mutex, size*size),
-		square:    make([][]byte, size*size),
-		squareSig: make(chan struct{}, 1),
-		squareDn:  make(chan struct{}),
-		span:      trace.SpanFromContext(ctx),
-	}
-
-	square, err := rsmt2d.ImportExtendedDataSquare(ses.square, ses.codec, ses.treeFn)
+	square, err := rsmt2d.ImportExtendedDataSquare(make([][]byte, size*size), DefaultRSMT2DCodec(), func() rsmt2d.Tree {
```

Comment:
Nit but I'd prefer the following for method calls with verbose args: 

```go
	emptySquare := make([][]byte, size*size)
	treeConstructorFn := func() rsmt2d.Tree {
		tree := wrapper.NewErasuredNamespacedMerkleTree(uint64(size)/2, nmt.NodeVisitor(adder.Visit))
		return &tree
	}	
	square, err := rsmt2d.ImportExtendedDataSquare(emptySquare, DefaultRSMT2DCodec(), treeConstructorFn)
	if err != nil {
		return nil, err
	}

```

Reply:
Updated